### PR TITLE
ufal/downloading-restricted-bitstreams-not-working-properly

### LINKED ISSUE #455: UFAL/Downloading of restricted bitstreams is not working properly
## Problem description
There are these bugs:
- [x] The admin must fill in required info from clarin license. It should download the bitstream.
- [x] The bitstream with ACA and RES licenses could be possible to download only for signed in users. Actually there is an wrong behaviour.
- [x] After the login the user is not redirected back to the file downloading.
- [x] This authorization is not working for downloading bitstreams on zip.

### PR DESCRIPTION
**Pull request by milanmajchrak**
| Phases            | MP | MM  |  MB  | MR |   JM | Total  |
|-----------------|----:|----:|----:|-----:|-----:|-------:|
| ETA                  |  0  |  0  |    0 |     0 |      0 |        0 |
| Developing      |  0  |  0  |    0 |    0 |      0 |         0 |
| Review             |  0  |  0  |    0 |    0 |      0 |         0 |
| Total                |   -  |   -  |   -   |  -    |   -    |         0 |
| ETA est.             |      |      |       |       |         |         0 |
| ETA cust.           |   -  |   -  |   -  |   -   |   -     |        0 |

### CODE REVIEW

File: src/app/item-page/item-page-routing.module.ts [hunk 1]
```diff
@@ -77,7 +80,17 @@ import { DSOEditMenuResolver } from '../shared/dso-page/dso-edit-menu.resolver';
...
-          }
+          },
+          {
+            path: 'download',
+            component: ClarinZipDownloadPageComponent,
+            resolve: {
+              dso: ItemPageResolver,
+            },
+            data: {
+              zipDownloadLink: 'This is downlaod link'
```
Comment (vidiecan) [inline src/app/item-page/item-page-routing.module.ts]: typo

File: src/app/bitstream-page/clarin-bitstream-download-page/clarin-bitstream-download-page.component.ts [hunk 2]
```diff
@@ -109,8 +113,15 @@ export class ClarinBitstreamDownloadPageComponent implements OnInit {
         } else {
           fileLink = isNotNull(this.dtoken) ? fileLink + '?dtoken=' + this.dtoken : fileLink;
         }
-        bitstreamURL = isNotNull(this.dtoken) ? bitstreamURL + '?dtoken=' + this.dtoken : bitstreamURL ;
+        bitstreamURL = isNotNull(this.dtoken) ? bitstreamURL + '?dtoken=' + this.dtoken : bitstreamURL;
+      }
+      if (isNotEmpty(this.zipDownloadLink.getValue())) {
+        const authToken = fileLink.substring(fileLink.indexOf('authentication-token'));
+        fileLink = this.zipDownloadLink.getValue() + '&' + authToken;
```
Comment (vidiecan) [inline src/app/bitstream-page/clarin-bitstream-download-page/clarin-bitstream-download-page.component.ts]: ?

File: src/app/item-page/simple/item-page.component.ts [hunk 3]
```diff
@@ -227,7 +227,7 @@ export class ItemPageComponent implements OnInit {
   }
 
   downloadFiles() {
-    window.location.href = this.halService.getRootHref() + `/core/bitstreams/allzip?handleId=${this.itemHandle}`;
+    void this.router.navigateByUrl(this.router.url + '/download');
```
Comment (vidiecan) [inline src/app/item-page/simple/item-page.component.ts]: ?

**Review by vidiecan — CHANGES_REQUESTED** [review]

File: src/app/bitstream-page/clarin-bitstream-download-page/clarin-bitstream-download-page.component.ts (hunk 2 above)
Comment (milanmajchrak) [reply src/app/bitstream-page/clarin-bitstream-download-page/clarin-bitstream-download-page.component.ts]: Fixed

File: src/app/item-page/simple/item-page.component.ts (hunk 3 above)
Comment (milanmajchrak) [reply src/app/item-page/simple/item-page.component.ts]: Fixed

File: src/app/item-page/item-page-routing.module.ts (hunk 1 above)
Comment (milanmajchrak) [reply src/app/item-page/item-page-routing.module.ts]: Fixed

**Pull request merged**